lib/ansible/modules/network/cloudengine/ce_vxlan_vap.py

```diff
@@ -0,0 +1,950 @@
+                self.updates_cmd.append(
+                    "undo bridge-domain %s" % self.bridge_domain_id)
+                xml_str = CE_NC_DELETE_BD_INTF % (
+                    self.bridge_domain_id, self.l2_sub_interface)
+
+        if not xml_str:
+            return
+        recv_xml = set_nc_config(self.module, xml_str)
+        self.check_response(recv_xml, "CONFIG_VAP_SUB_INTERFACE")
+        self.changed = True
+
+    def config_vap_vlan(self):
+        """configure a VLAN as a service access point"""
+
+        if not self.vap_info:
+            self.module.fail_json(msg="Error: Bridge domain does not exist.")
+
+        xml_str = ""
+        if self.state == "present":
+            if not is_vlan_in_bitmap(self.bind_vlan_id, self.vap_info["vlanList"]):
+                self.updates_cmd.append("bridge-domain %s" %
+                                        self.bridge_domain_id)
+                self.updates_cmd.append(
+                    "l2 binding vlan %s" % self.bind_vlan_id)
+                vlan_bitmap = vlan_vid_to_bitmap(self.bind_vlan_id)
+                xml_str = CE_NC_MERGE_BD_VLAN % (
+                    self.bridge_domain_id, vlan_bitmap, vlan_bitmap)
+        else:
+            if is_vlan_in_bitmap(self.bind_vlan_id, self.vap_info["vlanList"]):
+                self.updates_cmd.append("bridge-domain %s" %
+                                        self.bridge_domain_id)
+                self.updates_cmd.append(
+                    "undo l2 binding vlan %s" % self.bind_vlan_id)
+                vlan_bitmap = vlan_vid_to_bitmap(self.bind_vlan_id)
+                xml_str = CE_NC_MERGE_BD_VLAN % (
+                    self.bridge_domain_id, "0" * 1024, vlan_bitmap)
+
+        if not xml_str:
+            return
+        recv_xml = set_nc_config(self.module, xml_str)
+        self.check_response(recv_xml, "CONFIG_VAP_VLAN")
+        self.changed = True
+
+    def is_vlan_valid(self, vid, name):
+        """check vlan id"""
+
+        if not vid:
+            return
+
+        if not vid.isdigit():
+            self.module.fail_json(msg="Error: %s is not digit." % name)
+            return
+
+        if int(vid) < 1 or int(vid) > 4094:
+            self.module.fail_json(
+                msg="Error: %s is not in the range from 1 to 4094." % name)
+
+    def is_l2_sub_intf_valid(self, ifname):
+        """check l2 sub interface valid"""
+
+        if ifname.count('.') != 1:
+            return False
+
+        if_num = ifname.split('.')[1]
+        if not if_num.isdigit():
+            return False
+
+        if int(if_num) < 1 or int(if_num) > 4096:
+            self.module.fail_json(
+                msg="Error: Sub-interface number is not in the range from 1 to 4096.")
+            return False
+
+        if not get_interface_type(ifname):
+            return False
+
+        return True
+
+    def check_params(self):
+        """Check all input params"""
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```